pkg/apiserver/api_installer.go

```diff
@@ -49,16 +53,16 @@ func (a *APIInstaller) Install() (ws *restful.WebService, errors []error) {
 
 	// Initialize the custom handlers.
 	watchHandler := (&WatchHandler{
-		storage:                a.restHandler.storage,
-		codec:                  a.restHandler.codec,
-		canonicalPrefix:        a.restHandler.canonicalPrefix,
-		selfLinker:             a.restHandler.selfLinker,
-		apiRequestInfoResolver: a.restHandler.apiRequestInfoResolver,
+		storage: a.group.storage,
```

Comment:
Yeah, I think I'd like to refactor watch and redirect like the other methods, at which point group can be refactored out as well.


> On Feb 11, 2015, at 7:34 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/apiserver/api_installer.go:
> 
> > @@ -49,16 +53,16 @@ func (a *APIInstaller) Install() (ws *restful.WebService, errors []error) {
> >  
> >  	// Initialize the custom handlers.
> >  	watchHandler := (&WatchHandler{
> > -		storage:                a.restHandler.storage,
> > -		codec:                  a.restHandler.codec,
> > -		canonicalPrefix:        a.restHandler.canonicalPrefix,
> > -		selfLinker:             a.restHandler.selfLinker,
> > -		apiRequestInfoResolver: a.restHandler.apiRequestInfoResolver,
> > +		storage: a.group.storage,
> Why doesn't it make sense to just pass a.group, as opposed to the individual fields? Because they're all private, currently?
> 
> —
> Reply to this email directly or view it on GitHub.
>